index.js

```diff
@@ -542,11 +549,16 @@ SendStream.prototype.pipe = function pipe (res) {
   }
 
   // index file support
+  if (typeof this._index === 'function') {
```

Comment:
I would in fact remove the `index` event at all... This would be moved there too, but then the check for the function would need to be added both on the next `if` conditional AND inside the `sendIndex()` function too.